test/integration/authorization_test.go

```diff
@@ -0,0 +1,54 @@
+// +build integration,!no-etcd
+
+package integration
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+
+	policy "github.com/openshift/origin/pkg/cmd/experimental/policy"
+)
+
+func TestRestrictedAccessForProjectAdmins(t *testing.T) {
+	startConfig, err := StartTestMaster()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	openshiftClient, openshiftClientConfig, err := startConfig.GetOpenshiftClient()
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	// TODO remove once bootstrap authorization rules are tightened
+	removeInsecureOptions := &policy.RemoveGroupOptions{
+		RoleNamespace:    "master",
+		RoleName:         "cluster-admin",
+		BindingNamespace: "master",
+		Client:           openshiftClient,
+		Groups:           []string{"system:authenticated", "system:unauthenticated"},
+	}
+	if err := removeInsecureOptions.Run(); err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	haroldClient, err := CreateNewProject(openshiftClient, *openshiftClientConfig, "hammer-project", "harold")
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	markClient, err := CreateNewProject(openshiftClient, *openshiftClientConfig, "mallet-project", "mark")
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	_, err = haroldClient.Deployments("hammer-project").List(labels.Everything(), labels.Everything())
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	_, err = markClient.Deployments("hammer-project").List(labels.Everything(), labels.Everything())
+	if (err == nil) || (!strings.Contains(err.Error(), "Forbidden")) {
```

Comment:
can you use `kerrors.IsForbidden(err)`? I'm never sure what type of error we're getting back at each layer